### PR TITLE
Various fixes and features for dealing with realistic data.

### PR DESCRIPTION
**Pull request by tskisner**
* Move HWPSS utility functions into a separate source file

* Add a new CalibrateDetectors operator which takes a dictionary of factors to apply per observation.

* Change detector timeconstant deconvolution to use serial rather than batched FFTs by default.  This reduces memory footprint in the common case where most parallelism comes from MPI.

* Add option to AzimuthIntervals to also cut extraneous long intervals

* Support both PDF and PNG image formats for plots.

* When demodulating data, also propagate per-detector flags.

* Fix deadlocks caused by logging barriers in noise estimation.

### CODE REVIEW

**Review by keskitalo — APPROVED** [review]
Looks good. Please look over the few comments and decide for yourself if any are actionable.

File: src/toast/hwp_utils.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    good = my_flags == 0
+    # Compute local upper triangle
+    for hr in range(0, n_harmonics):
+        for hc in range(hr, n_harmonics):
+            my_cov[4 * hr + 0, 4 * hc + 0] = np.dot(
+                my_sincos[good, 2 * hr + 0], my_sincos[good, 2 * hc + 0]
+            )
+            my_cov[4 * hr + 0, 4 * hc + 1] = np.dot(
+                my_sincos[good, 2 * hr + 0],
+                np.multiply(my_times[good], my_sincos[good, 2 * hc + 0]),
```
[the remark this answers is not in the document]
Comment (tskisner) [reply src/toast/hwp_utils.py:64]: Good point, this existing code was moved out of the hwpss template, but I should expand the docstrings and link to references.  This is implementing the technique used for Maxipol / EBEX which has time drift coefficients for each harmonic.

File: src/toast/hwp_utils.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    if comm is None:
+        return my_sincos
+    sincos = np.vstack(comm.allgather(my_sincos))
+    return sincos
+
+
+def hwpss_compute_coeff_covariance(times, flags, sincos, comm=None):
+    slc = hwpss_samples(len(times), comm)
+    my_sincos = sincos[slc, :]
+    my_times = times[slc]
```
[the remark this answers is not in the document]
Comment (tskisner) [reply src/toast/hwp_utils.py:50]: I should expand the docs.  Basically the calling code is passing in timestamps that are relative to the observation start.  So these are not giant numbers.

File: src/toast/ops/noise_model.py [hunk 3]
```diff
@@ -518,27 +518,26 @@ def _fit_log_psd(self, freqs, data, guess=None):
         bad = np.logical_not(good)
         n_bad = np.count_nonzero(bad)
         if n_bad > 0:
+
```
[the remark this answers is not in the document]
Comment (tskisner) [reply src/toast/ops/noise_model.py]: Fixed.

File: src/toast/hwp_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tskisner) [reply src/toast/hwp_utils.py:133]: Ok, docstrings in place, including discussion of the algorithm and link to Joy's thesis which has a more detailed treatment.

**Pull request merged**